test/test_common.py

```diff
@@ -28,7 +32,11 @@ class TestCommon(unittest.TestCase):
     """
 
     def _check(self, tensor, reference, msg):
-        test_passed = (tensor == reference).all().item() == 1
+        if tensor.dtype == torch.long:
+            tensor = tensor.float()
+        if reference.dtype == torch.long:
+            reference = reference.float()
+        test_passed = torch.allclose(tensor, reference, rtol=1/2 ** 15)
```

Comment:
Q: Does this look strange? (to use ```rtol``` with 1/2 ** 15)
I think this might also be changed to ```1e-4```<-- this is the biggest power of 10 that we could use to "catch" the precision given by 2**16 (< 10^5)